test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -1293,6 +1297,210 @@ public void Process_GeneratesFormattedOutput_ForDateTime(string specifiedType, s
+            string propertyName,
+            string expectedValue,
+            string expectedType)
+        {
+            // Arrange
+            var expectedAttributes = new TagHelperAttributeList
+            {
+                { "type", expectedType },
+                { "id", propertyName },
+                { "name", propertyName },
+                { "value", expectedValue },
+            };
+
+            var context = new TagHelperContext(
+                tagName: "input",
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+
+            var output = new TagHelperOutput(
+                expectedType,
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(
+                    new DefaultTagHelperContent()))
+            {
+                TagMode = TagMode.SelfClosing,
+            };
+
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var model = new DateTime(
+                year: 2000,
+                month: 1,
+                day: 2,
+                hour: 3,
+                minute: 4,
+                second: 5,
+                millisecond: 60,
+                kind: DateTimeKind.Utc);
+
+            var htmlGenerator = HtmlGeneratorUtilities.GetHtmlGenerator(metadataProvider);
+            var tagHelper = GetTagHelper(
+                htmlGenerator,
+                model: model,
+                propertyName: propertyName,
+                metadataProvider: metadataProvider);
+            tagHelper.ViewContext.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
+
+            // Act
+            await tagHelper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.Equal(expectedAttributes, output.Attributes);
+            Assert.Equal(expectedType, output.TagName);
+        }
+
+        // Html5DateRenderingMode.Rfc3339 can be disabled.
+        [Theory]
+        [InlineData("Date", null, "02/01/2000", "date")]
+        [InlineData("Date", "{0:d}", "02/01/2000", "date")]
+        [InlineData("DateTime", null, "02/01/2000 03:04:05", "datetime-local")]
+        [InlineData("DateTimeLocal", null, "02/01/2000 03:04:05", "datetime-local")]
+        [InlineData("DateTimeOffset", null, "02/01/2000 03:04:05", "text")]
+        [InlineData("DateTimeOffset", "{0:o}", "2000-01-02T03:04:05.0600000Z", "text")]
+        [InlineData("Time", null, "03:04", "time")]
+        [InlineData("Time", "{0:t}", "03:04", "time")]
+        [InlineData("Month", null, "2000-01", "month")]
+        [InlineData("Month", "{0:yyyy/MM}", "2000/01", "month")]
+        [InlineData("Week", null, "1999-W52", "week")]
+        [InlineData("Week", "{0:yyyy/MM}", "2000/01", "week")]
```

Comment:
Matching the custom format for `"month"` looks odd. Suggest using a literal e.g. `"{0:yyyy/'W53'}"`.